python/ray/experimental/workflow/workflow_mananger.py

```diff
@@ -0,0 +1,180 @@
+import functools
+from typing import List, Tuple, Union, Any, Dict, Callable, Optional
+import uuid
+
+import ray
+
+from ray.experimental.workflow import workflow_context
+
+RRef = ray.ObjectRef  # Alias ObjectRef because it is too long in type hints
+WorkflowOutputType = Union[RRef, Tuple[RRef]]
+StepArgType = Union[RRef, Any]
+RayTaskOutputType = Union[RRef, List[RRef], Tuple[RRef]]
+
+
+def resolve_object_ref(ref: RRef) -> Tuple[Any, RRef]:
+    """
+    Resolves the ObjectRef into the object instance.
+
+    Returns:
+        The object instance and the direct ObjectRef to the instance.
+    """
+    assert ray.is_initialized()
+    last_ref = ref
+    while True:
+        if isinstance(ref, RRef):
+            last_ref = ref
+        else:
+            break
+        ref = ray.get(last_ref)
+    return ref, last_ref
+
+
+def wrap_step_outputs(outputs: RayTaskOutputType) -> Tuple[WorkflowOutputType, List[str]]:
+    """
+    This function converts returned object refs to workflow
+    step results. It also collects hex strings of these refs.
+
+    Args:
+        outputs: The outputs of a Ray remote call.
+
+    Returns:
+        Corresponding WorkflowStepResult objects and
+        related hex string.
+    """
+    if isinstance(outputs, RRef):
+        results = [outputs]
+    elif isinstance(outputs, tuple):
+        results = outputs
+    else:
+        raise TypeError("Unknown return type")
+    output_ids = [o.hex() for o in results]
+    return outputs, output_ids
```

Comment:
I'm a little bit confused here. It looks like WorkflowOutputType and RayTaskOutputType are actually the same. (except the List which is not taken care of here).